lib/rubocop/cop/rspec/rails/inferred_spec_type.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Rails
+        # Identifies redundant spec type.
+        #
+        # @safety
+        #   This cop is marked as unsafe because
+        #   `config.infer_spec_type_from_file_location!` may not be enabled.
+        #
+        # @example
+        #
+        #   # bad
+        #   # spec/models/user_spec.rb
+        #   RSpec.describe User, type: :model
+        #
+        #   # good
+        #   # spec/models/user_spec.rb
+        #   RSpec.describe User
+        #
+        #   # good
+        #   # spec/models/user_spec.rb
+        #   RSpec.describe User, type: :request
+        #
+        class InferredSpecType < Base
+          extend AutoCorrector
+
+          # @return [Array<Hash>]
+          INFERENCES = [
```

Comment:
I'd make this configurable so that the cop could detect a redundant `type: :service` in `spec/service/` specs.